sensing/autoware_radar_objects_adapter/src/radar_objects_adapter.cpp

```diff
@@ -263,8 +263,18 @@ void RadarObjectsAdapter::parse_as_detections(
   for (const auto & input_object : input_msg.objects) {
     autoware_perception_msgs::msg::DetectedObject output_object;
 
+    // Populate common fields
     const auto & yaw = input_object.orientation;
     populate_common_fields(input_object, output_object, yaw);
+
+    // Set flags for kinematics
+    output_object.kinematics.has_position_covariance = true;
+    output_object.kinematics.orientation_availability =
+      autoware_perception_msgs::msg::DetectedObjectKinematics::AVAILABLE;
+    output_object.kinematics.has_twist = true;
+    output_object.kinematics.has_twist_covariance = true;
```

Comment:
[just a comment]
This part could highly depend on original radar message definition, this is now correct for our continental radar but may change for the future.